docs/source/name_inference.rst

```diff
@@ -0,0 +1,426 @@
+.. currentmodule:: torch
+
+.. _name_inference_reference-doc:
+
+Named Tensors operator coverage
+==============================
+
+.. warning::
+    The named tensor API is experimental and subject to change.
+
+Named tensors use names to check that some APIs are being used correctly and
+propagate names in a process called name inference.
+Name inference works as the following:
+
+- Check names: an operator may check that certain dimensions must match.
+- Propagate names: name inference computes and propagates names to output tensors.
+
+Below is a list of all operations that are supported with named tensors
+and their name inference rules. If an operation is not here then it is probably not
+supported but one can try to run the operation and see what happens.
+
+.. csv-table:: Supported Operations
```

Comment:
is there any way this table can be generated or can refer to some other list? I think people may forget to add operators as we build out support, leaving this incorrect.

Reply:
It's generated from a csv file that I keep locally. In the long term view of things we will get rid of the table as more operations support named tensors and directly inline name propagation rules into the API references for each operator. 

In the short term, I agree it would be nice to auto generate this table from some source, maybe even combining it with named tensor testing in `test_namedtensor.py`. That task is beyond the scope of this PR, though and would require some solid legwork.